esmvalcore/_recipe_checks.py

```diff
@@ -138,3 +139,19 @@ def extract_shape(settings):
                 f"In preprocessor function `extract_shape`: Invalid value "
                 f"'{value}' for argument '{key}', choose from "
                 "{}".format(', '.join(f"'{k}'".lower() for k in valid[key])))
+
+
+def extract_levels(settings, variable):
+    short_name = variable['short_name']
+    table_entry = CMOR_TABLES[variable['cmor_table']].get_variable(
+        variable['mip'],
+        short_name,
+        variable.get('derive', False),
+    )
+    for coordinate in table_entry.coordinates.values():
+        if coordinate.axis == 'Z':
+            break
```

Comment:
careful here with generic coordinates like `olevel` or `alevel` -> are they mapped to eg coord name `plev` at this stage? Can't remember @jvegasbsc